pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/ColumnPartitionMetadata.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.metadata.segment;
+
+import com.linkedin.pinot.common.config.ColumnPartitionConfig;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.math.IntRange;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.codehaus.jackson.map.annotate.JsonDeserialize;
+import org.codehaus.jackson.map.annotate.JsonSerialize;
+
+
+/**
+ * Class for partition related column metadata:
+ * <ul>
+ *   <li> Partition function.</li>
+ *   <li> Number of partitions. </li>
+ *   <li> List of partition ranges. </li>
+ * </ul>
+
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ColumnPartitionMetadata extends ColumnPartitionConfig {
```

Comment:
That will duplicate a lot of code. Let me think about how to address that (in a separate PR).